### PR TITLE
Categorize, BOM, dependabot, and new Jenkins base version

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Categorize, BOM, Dependabot, and new Jenkins base version

* Categorize the plugin in "System Configuration" 
* Require Jenkins 2.263.1 or newer (so we can use the bom)
* Use plugin bom to provide dependency versions
* Use documentation as code for plugin docs
* Configure dependabot to update dependencies

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Implied Labels Plugin
+
+Infer dynamic node labels using configured implications.
+
+Lets have a node with `linux`, `rhel`, `rhel6` and `rhel6_4` labels.
+Most of this information is redundant and can be inferred from `rhel6_4` label.
+Here is an excerpt of Implied Labels Plugin configuration to do just that.
```
Comment (MarkEWaite) [inline README.md:7]: Use a current Red Hat Enterprise Linux version in the example.
```suggestion
Lets have a node with `linux`, `rhel`, `rhel8` and `rhel8_4` labels.
Most of this information is redundant and can be inferred from `rhel8_4` label.
Here is an excerpt of Implied Labels plugin configuration to do just that.
```

File: pom.xml [hunk 2]
```diff
@@ -3,34 +3,45 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.55</version>
+    <version>4.19</version>
```
Comment (MarkEWaite) [inline pom.xml:6]: Use the newest parent pom
```suggestion
    <version>4.24</version>
```

File: pom.xml [hunk 3]
```diff
@@ -3,34 +3,45 @@
...
+    <url>https://github.com/${gitHubRepo}</url>
+    <tag>${scmTag}</tag>
   </scm>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.jenkins.tools.bom</groupId>
+        <artifactId>bom-2.263.x</artifactId>
+        <version>887.vae9c8ac09ff7</version>
```
Comment (MarkEWaite) [inline pom.xml:29]: Use the newest Jenkins plugin BOM.
```suggestion
        <version>937.v51fde92016ed</version>
```

**Review by MarkEWaite — COMMENTED** [review]
Additional changes to propose

**Comment by MarkEWaite** [comment]
@olivergondza I've rebased the changes and made sure that they are well formed commits with justification for each change.

**Comment by olivergondza** [comment]
Thanks, Mark!

**Pull request merged**